File: platforms/kotlin/examples/scene/light/set_range.kt
import org.fragmentcolor.*

val lamp = Light.point(listOf(0.0f, 2.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
lamp.setRange(8.0)
val negative = lamp.setRange(-1.0)

// Directional lights have no range — the call errors.
val sun = Light.directional(listOf(0.0f, -1.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
val unsupported = sun.setRange(5.0)

File: platforms/python/examples/scene/light/set_range.py
from fragmentcolor import Light

lamp = Light.point([0.0, 2.0, 0.0], [1.0, 1.0, 1.0])
lamp.set_range(8.0)
negative = lamp.set_range(-1.0)

# Directional lights have no range — the call errors.
sun = Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0])
unsupported = sun.set_range(5.0)